openstack/migrate-helpers/osmigrator.rb

```diff
@@ -0,0 +1,193 @@
+        if group["name"] == "default"
+          next
+        end
+
+        secgroup_name = group["name"]
+
+        group_details = JSON.parse(`neutron security-group-show #{secgroup_name} -f json`)
+
+        importer_script.puts("# Start of #{secgroup_name} Rules")
+
+        # Add the command to create the security group.
+        sec_description = group_details[0]["Value"]
+        if sec_description.nil?
+          importer_script.puts("neutron security-group-create #{secgroup_name}")
+        else
+          importer_script.puts("neutron security-group-create #{secgroup_name} --description #{sec_description}")
+        end
+
+        # Parse out the actual rules. It requires a little bit of hackery because its not actually json.
+        rules = JSON.parse('[' + group_details[3]["Value"].gsub(/}\n/, "},") + ']')
+
+        rules.each do |rule|
+          #$ neutron security-group-rule-create --direction ingress --ethertype IPv4 --protocol tcp --port-range-min 80 --port-range-max 80 --remote-ip-prefix 0.0.0.0/0 global_http
+          importer_script.puts("neutron security-group-rule-create \\")
+          rule.each do |key, val|
+            unless val.nil?
+              opt = key.gsub(/_/, "-")
+              importer_script.puts("        --#{opt} #{val} \\")
+            end
+          end
+          importer_script.puts("        #{secgroup_name}")
+        end
+
+        importer_script.puts("# End of #{secgroup_name} Rules\n\n")
+
+      end
+    end
+  end
+
+  def node_export
+    CSV.open("#{OUTPUT_FOLDER}/#{FILE_NAMES[:nodes]}", "w",
+             :col_sep => '|',
+             :write_headers => true,
+             :headers => ["node_name", "flavor-image", "raw-flavor", "raw-image", "security_groups"]) do |export|
+      all_nodes = `nova list | grep SERVICE | cut -d'|' -f3 | sort`.split("\n").map{|n| n = n.strip }
+      all_nodes.each do |node|
+        # [flavor, image, security_groups]
+        node_details = `nova show #{node} | grep -E '(flavor|image|security_groups)' | cut -d'|' -f3`
+        node_details = node_details.split("\n").map{ |r| r = r.split("(")[0].strip }
+
+        # ["m1.medium", "trusty-image"] -> "trusty.medium"
+        flavor_image = node_details[1].split('-')[0] + "." + node_details[0].split(".")[1]
+
+        export << [node, flavor_image, node_details[0], node_details[1], node_details[2]]
+      end
+    end
+  end
+
+  def flavor_export
+    flavors = `nova flavor-list`.split("\n")[3..-2]
+    # Clean up terminal output
+    flavors = flavors.map{ |f|
+      f = f.split("|").map{ |d|
+        d = d.strip()
+      }
+    }
+
+    CSV.open("#{OUTPUT_FOLDER}/#{FILE_NAMES[:flavors]}", "w",
+            :write_headers => true,
+            :headers => ["name", "id", "ram", "disk", "vcpus", "swap"]) do |export|
+      flavors.each do |flavor|
+        export << [flavor[2], flavor[1], flavor[3], flavor[4], flavor[7], flavor[6]]
+      end
+    end
+  end
+
+  def create_output_folder!
```

Comment:
Methods with a bang usually imply that they will raise an exception if they fail, so I think we should either remove the bang here, or not rescue.